jasmin_cloud/provider/openstack/provider.py

```diff
@@ -646,14 +652,18 @@ def delete_machine(self, machine):
         """
         See :py:meth:`.base.ScopedSession.delete_machine`.
         """
-        machine = machine.id if isinstance(machine, dto.Machine) else machine
-        self._log("Deleting machine '%s'", machine)
+        machine = machine if isinstance(machine, dto.Machine) else self.find_machine(machine)
+        self._log("Deleting machine '%s'", machine.id)
         # First, delete any associated ports
-        for port in self._connection.network.ports.all(device_id = machine):
-            port._delete()
-        self._connection.compute.servers.delete(machine)
         try:
```

Comment:
Not sure why this `try/except` block would be required? Given the changes you made to `_from_api_server`, the `provisioned_by_caas` attribute should always be populated...

In which case, `if not machine.provisioned_by_caas:` should be sufficient.

Reply:
Ah I thought that would be the case but wasn't certain so thought it would be safer to do put a try catch in